src/pubsub.c

```diff
@@ -285,10 +284,8 @@ int pubsubUnsubscribeChannel(client *c, robj *channel, int notify, pubsubtype ty
         de = dictFind(*type.serverPubSubChannels, channel);
         serverAssertWithInfo(c,NULL,de != NULL);
         clients = dictGetVal(de);
-        ln = listSearchKey(clients,c);
-        serverAssertWithInfo(c,NULL,ln != NULL);
-        listDelNode(clients,ln);
-        if (listLength(clients) == 0) {
+        serverAssertWithInfo(c,NULL,dictDelete(clients, c) == DICT_OK);
```

Comment:
As you are updating the line, we could add space after `,`
```suggestion
        serverAssertWithInfo(c, NULL, dictDelete(clients, c) == DICT_OK);
```